Mathlib/Algebra/Category/ModuleCat/Sheaf/ChangeOfRings.lean

```diff
@@ -36,3 +37,29 @@ noncomputable def restrictScalars :
   map φ := { val := (PresheafOfModules.restrictScalars α.val).map φ.val }
 
 end SheafOfModules
+
+namespace PresheafOfModules
+
+variable {R R' : Cᵒᵖ ⥤ RingCat.{u}} (α : R ⟶ R')
+  {M₁ M₂ : PresheafOfModules.{v} R'} (hM₂ : Presheaf.IsSheaf J M₂.presheaf)
+  [Presheaf.IsLocallySurjective J α]
+
+/-- The functor `PresheafOfModules.restrictScalars α` induces bijection on
+morphisms if `α` is locally surjective and the target presheaf is a sheaf. -/
+noncomputable def restrictHomEquivOfIsLocallySurjective :
+    (M₁ ⟶ M₂) ≃ ((restrictScalars α).obj M₁ ⟶ (restrictScalars α).obj M₂) where
+  toFun f := (restrictScalars α).map f
+  invFun g :=
+    { hom := g.hom
+      map_smul := fun X r' m => by
+        apply hM₂.isSeparated _ _ (Presheaf.imageSieve_mem J α r')
+        rintro Y p ⟨r : R.obj _, hr⟩
+        erw [M₂.map_smul, ← NatTrans.naturality_apply g.hom p.op m,
+          ← hr, ← g.map_smul _ r (M₁.presheaf.map p.op m),
+          ← NatTrans.naturality_apply g.hom p.op (r' • m),
+          M₁.map_smul p.op r' m, ← hr]
+        rfl }
+  left_inv _ := rfl
+  right_inv _ := rfl
```

Comment:
Should this use the new `fullyFaithful` API?

Reply:
No, we cannot, because there is an extra assumption on the target object (it has to be a sheaf).